src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -309,21 +343,105 @@ private FormValidation checkAgentConnectivity(final String host, final String po
             throws IOException, ServletException {
-        if (DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, Secret.fromString(targetApiKey))) {
+
+        Boolean validApiConnection = false;
+        StringCredentials credentialsApiKey = this.getCredentialFromId(targetCredentialsApiKey);
+        if (credentialsApiKey != null && !credentialsApiKey.getSecret().getPlainText().isEmpty()) {
+            validApiConnection = DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, credentialsApiKey.getSecret());
+        } else {
+            validApiConnection = DatadogHttpClient.validateDefaultIntakeConnection(targetApiURL, Secret.fromString(targetApiKey));
+        }
+        if (validApiConnection) {
             return FormValidation.ok("Great! Your API key is valid.");
         } else {
             return FormValidation.error("Hmmm, your API key seems to be invalid.");
         }
     }
 
+    /**
+     * Populates the targetCredentialsApiKey field from the configuration screen with all of the valid credentials
+     *
+     * @param item - The context within which to list available credentials
+     * @param targetCredentialsApiKey - A String containing the API key as a credential
+     * @return a ListBoxModel object used to display all of the available credentials.
+     */
+    public ListBoxModel doFillTargetCredentialsApiKeyItems(
+        @AncestorInPath Item item,
+        @QueryParameter String targetCredentialsApiKey
+        ) {
+        StandardListBoxModel result = new StandardListBoxModel();
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return result.includeCurrentValue(targetCredentialsApiKey);
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ)
+                && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return result.includeCurrentValue(targetCredentialsApiKey);
+            }
+        }
+        return result.includeEmptyValue()
+            .includeMatchingAs(ACL.SYSTEM,
+                               Jenkins.get(),
+                               StringCredentials.class,
+                               Collections.emptyList(),
+                               CredentialsMatchers.instanceOf(StringCredentials.class))
+            .includeCurrentValue(targetCredentialsApiKey);
+    }
+
+
+    /**
+     * Tests the targetCredentialsApiKey field from the configuration screen, to check its' validity.
+     *
+     * @param item - The context within which to list available credentials.
+     * @param targetCredentialsApiKey - A String containing the API key as a credential
+     * @return a FormValidation object used to display a message to the user on the configuration
+     * screen.
+     */
+    @RequirePOST
+    public FormValidation doCheckCredentialsApiKey(
+        @AncestorInPath Item item,
+        @QueryParameter String targetCredentialsApiKey
```

Comment:
Don't you need to add the name of the param?
```suggestion
        @QueryParameter("targetCredentialsApiKey") String targetCredentialsApiKey
```

Reply:
Added!